File: tests/conformance/fixtures/elixir/24-success-criteria.exs
use Sykli

pipeline do
  task "test" do
    run("go test ./...")
    task_type(:test)

    success_criteria([
      exit_code(0),
      file_exists("coverage.out")
    ])
  end

  task "package" do
    run("go build -o dist/app ./...")
    task_type(:package)

    success_criteria([
      file_non_empty("dist/app")
    ])

    after_(["test"])
  end
end

File: tests/conformance/fixtures/python/24-success-criteria.py
from sykli import Pipeline

p = Pipeline()
p.task("test").run("go test ./...").task_type("test").success_criteria([
    {"type": "exit_code", "equals": 0},
    {"type": "file_exists", "path": "coverage.out"},
])
p.task("package").run("go build -o dist/app ./...").task_type("package").success_criteria([
    {"type": "file_non_empty", "path": "dist/app"},
]).after("test")
p.emit()
